### PR TITLE
runners: Black Magic Probe runner

### PR DESCRIPTION
**Pull request by ntavish**
Based on https://github.com/zephyrproject-rtos/zephyr/pull/6829 by @diytronic, updated and added
debug and attach commands.

### CODE REVIEW

File: src/west/runners/blackmagicprobe.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    def bmp_debug(self, command, **kwargs):
+        command = (self.gdb +
+            ['-ex', "set confirm off",
+            '-ex', "target extended-remote {}".format(self.gdb_serial),
+            '-ex', "monitor swdp_scan",
+            '-ex', "attach 1",
+            '-ex', "file {}".format(self.kernel_elf),
+            '-ex', "load {}".format(self.kernel_elf),
+            '-ex', "break main",
```
Comment (mbolivar) [inline src/west/runners/blackmagicprobe.py]: None of the other runners break on main - can you try to match their behavior for consistency?

File: src/west/runners/blackmagicprobe.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    def do_run(self, command, **kwargs):
+
+        if command == 'flash':
+            self.bmp_flash(command, **kwargs)
+        elif command == 'debug':
+            self.bmp_debug(command, **kwargs)
+        elif command == 'attach':
+            self.bmp_attach(command, **kwargs)
+        else:
+            self.bmp_flash(command, **kwargs)
```
Comment (mbolivar) [inline src/west/runners/blackmagicprobe.py]: Please add newline at eof

File: src/west/runners/blackmagicprobe.py (hunk 1 above)
Comment (ntavish) [reply src/west/runners/blackmagicprobe.py]: fixed

**Comment by mbolivar** [comment]
Looks good to me.

Feel free to add test cases as a follow up commit.

**Pull request merged**